test/argo.misc.test.ts

```diff
@@ -43,38 +47,70 @@ describe("argo-cd-bot", () => {
     afterEach(() => {
         sandbox.restore()
         nock.cleanAll()
+        nock.enableNetConnect();
     })
 
-    test("help comment posted on PR", async() => {
+    test("help comment posted on opened PR", async () => {
         nock("https://api.github.com")
             .post("/app/installations/2/access_tokens")
-            .reply(200, {token: "test"})
+            .reply(200, { token: "test" })
 
         // test constants
         const branch = "newBranch"
         const appDiff = "===== App Diff ===="
         const appName = "app1"
         const appDir = "projects/app1"
+        const commentBody = "hello world from argocd bot!"
 
-        nock("https://api.github.com").get("/repos/robotland/test/pulls").reply(200, {"data": {"number": 109, "head": { "ref": branch}}})
+        nock("https://api.github.com")
+            .get("/repos/robotland/test/pulls")
+            .reply(200, { "data": { "number": 1, "head": { "ref": branch } } })
+
+        // regex match post body should match diff produced by API
+        let prCommentResponse = nock("https://api.github.com")
+            .post("/repos/robotland/test/issues/1/comments", (response: { body: string; }) => {
+                expect(response.body).toMatch(prOpenedComment)
```

Comment:
The way i read this test is:

post to this API, and expect the response to `prOpenedComment` which is just an imported file? Can you explain this test to me a little more?

This comment might seem very pedantic so I apologize but i'm trying to learn how to write test in the future and just want to understand what's going one

Reply:
The way `nock` works is that it intercepts requests (so they never go to github) and ensures that the content of the request (or response, it can work both ways) is what is expected. So in this case, it's mocking out the Github api. So the flow for this test starts with `probot.receive`, which tests receiving a webhook from github. The test ensures that the pull request opened handler is properly registered with the probot, and then ensures that the response to the webhook has the expected result. `nock` will also ensure that the request was invoked, meaning, if the argo bot never responded, or responded to the wrong uri or with the wrong content, the test will fail.

Reply:
I forgot to mention that the important statement here is the `prCommentRequest.done()` since that is what does the evaluation of `expect(response.body).toMatch(prOpenedComment)`. This happens after `probot.receive` so that the workflow is exercised before we assert that the api was invoked.